Assets/Scripts/PlayerMovement.cs

```diff
@@ -17,9 +19,12 @@ void Update()
     {
         moveInput = Input.GetAxis("Horizontal"); // if I press right key moveInput = 1,left key = -1 
         ballRigidBody.velocity = new Vector2(moveInput * speed, ballRigidBody.velocity.y); // adds force to the ball movement in order for the player to move,doesn't affect y axis
-        if (Input.GetKeyDown(KeyCode.Space) && grounded) // checks if jumping key is pressed
+        if (grounded)
+            remainingJumps = maxJumps; // if the player is grounded the remaningJumps variable is equal to the maxJum
```

Comment:
you can move this part of code in the `void OnCollisionEnter2D(Collision2D collision)` function because here you verify if the ball touched the ground and then you reset the counter but in the above function you already test this.
You can write      
```
    void OnCollisionEnter2D(Collision2D collision) // function to detect when object enters in collision with ground
    {
        if (collision.gameObject.tag == "ground")
        {
            grounded = true;
            remainingJumps = maxJumps; // if the player is grounded the remainingJumps variable is equal to the maxJumps
        }
    }

```

and remove
```
        if (grounded)
            remainingJumps = maxJumps; // if the player is grounded the remaningJumps variable is equal to the maxJum

```